### PR TITLE
This is supposed to enable Nested objects to be used as grid data

### PR DESCRIPTION
**Pull request by Cubelaster**
Basically added a bunch of .split('.') and foreach inside of existing methods for handling sorts/groupings and other grid manipulation methods. 
Don't know if there are more places that this needs to be added. 
I tried with 200k rows and it showed no delay.

### CODE REVIEW

File: src/utilities/array.ts [hunk 1]
```diff
@@ -56,7 +63,22 @@ export const sortArray = (inputArray: Array<any>, sortOptions: Array<SortProps>)
 
 export const groupBy = function (inputArray, groupProp) {
     return inputArray.reduce((groups, item) => {
-        (groups[item[groupProp]] = groups[item[groupProp]] || []).push(item);
+        let result = item;
+        let splits = groupProp.split('.');
+        for (let i = 0; i < splits.length; i++) {
```
Comment (vinkobedek) [inline src/utilities/array.ts]: won't this for loop just set the result to the deepest value?

File: src/utilities/array.ts [hunk 2]
```diff
@@ -38,8 +38,15 @@ export const sortArray = (inputArray: Array<any>, sortOptions: Array<SortProps>)
                 valueA = sortOption.sortFunction(a);
                 valueB = sortOption.sortFunction(b);
             } else {
-                valueA = a[sortOption.column];
-                valueB = b[sortOption.column];
+                let splits = sortOption.column.split('.');
+                let resultA = a;
+                let resultB = b;
+                for (let i = 0; i < splits.length; i++) {
```
Comment (vinkobedek) [inline src/utilities/array.ts]: same as for the groupinng operation, won't this just set valueA and ValueB to the deepest value?

File: src/components/QuickGrid/DataSelectors.ts [hunk 3]
```diff
@@ -79,22 +79,24 @@ const filterRows = (rows: Array<any>, columns: Array<GridColumn>, searchText: st
...
-
     let members = columns.map(col => col.valueMember);
-
     const newRows = rows.filter(row => {
         let visible = false;
         for (let value of members) {
-            if (row[value].toString().toLowerCase().indexOf(filterText) !== -1) {
-                visible = true;
-                break;
+            let objectProps = value.split('.');
```
Comment (vinkobedek) [inline src/components/QuickGrid/DataSelectors.ts]: how's the performance impact of these changes?

File: src/utilities/array.ts (hunk 1 above)
Comment (Cubelaster) [reply src/utilities/array.ts]: It will. Can it be done better?

File: src/components/QuickGrid/DataSelectors.ts (hunk 3 above)
Comment (Cubelaster) [reply src/components/QuickGrid/DataSelectors.ts]: Switched from 100k to 200k and no impact.

File: src/utilities/array.ts (hunk 2 above)
Comment (Cubelaster) [reply src/utilities/array.ts]: Deepest FOUND.

File: src/utilities/array.ts (hunk 1 above)
Comment (Cubelaster) [reply src/utilities/array.ts]: I saw a method written by @hrestakd `getObjectValue` so I will try to use that instead

Comment (vinkobedek) [reply src/utilities/array.ts]: ah, sorry, my bad, now that i am looking at this again i see what you are actually doing here. All good.

Comment (hrestakd) [reply src/utilities/array.ts]: the method itself was really fast but the sorting for tree grid was the slow part so I removed it

Comment (hrestakd) [reply src/utilities/array.ts]: For QuickGrid purposes it has little to no performance impact

**Comment by Cubelaster** [comment]
Now switched to getObjectValue(item, prop);
http://take.ms/E3AUu
Performance video. 
This seems to make a hit on performance, but seems ok?

**Pull request merged**